src/internal/io.js

```diff
@@ -16,29 +16,14 @@ const FLUSH  = 'FLUSH'
 
 const effect = (type, payload) => ({[IO]: true, [type]: payload})
 
-export function take(channel, pattern) {
-  if(arguments.length >= 2) {
-    check(channel, is.notUndef, 'take(channel, pattern): channel is undefined')
-    check(channel, is.take, `take(channel, pattern): argument ${String(channel)} is not a valid channel (channel argument must have a take method)`)
-    check(pattern, is.notUndef, 'take(channel, pattern): pattern is undefined')
-    check(pattern, is.pattern, `take(channel, pattern): argument ${String(pattern)} is not a valid pattern (pattern must be String | Function: a => boolean | Array<String>)`)
-  } else if(arguments.length === 1) {
-    check(channel, is.notUndef, 'take(patternOrChannel): undefined argument')
-    if(!is.take(channel)) {
-      if(is.pattern(channel)) {
-        pattern = channel
-        channel = null
-      } else {
-        throw new Error(`take(patternOrChannel): argument ${String(channel)} is not valid channel or a valid pattern`)
-      }
-    } else {
-      pattern = '*'
-    }
-  } else {
-    pattern = '*'
+export function take(patternOrChannel = '*') {
```

Comment:
Oh, missed that as I've focused on checking if default argument has an impact on `arguments.length`. Could change `check(patternOrChannel, ...` to `check(arguments[0], ...`, but dont know how much you like such solution.

In previous version of `take` was defaulting too to `*`, so I just wanted to keep that behaviour cleanly in the code and it seems better to make it a default parameter than to override argument later. 

Just checked againt behaviour for some scenarios:
```
// function test(a) {}
test() // arguments.length === 0
test(undefined) // arguments.length === 1

// function test2(a = '*') {}
test() // arguments.length === 0, a === '*'
test(undefined) // arguments.length === 1, a === '*'
```